dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java

```diff
@@ -286,22 +359,32 @@ private File writeCleanupFile(int platformType) throws IOException {
     @Override
     public void run() {
         try {
-
             stopServer(); // first, stop server
-
             // When the server is launched with java -jar, delete the server on exit minus
             // the /logs folder, unless WLP_JAR_EXTRACT_DIR is set at which point don't delete
             // anything.
-
             if (extractDirPredefined != true) {
                 startAsyncDelete(); // now launch async process to cleanup extraction directory
             }
 
         } catch (Exception e) {
+            try {
+                Files.write(getHookLog(), (e.getMessage() + "\n" + Arrays.toString(e.getStackTrace())).getBytes(), StandardOpenOption.APPEND);
+            } catch (IOException e1) {
+                e1.printStackTrace();
+            }
             e.printStackTrace();
             throw new RuntimeException("Shutdown hook failed with exception " + e.getMessage());
         }
 
     }
 
+    /**
+     * @return
+     */
+    private Path getHookLog() {
+        String logDir = dir + File.separator + "wlp" + File.separator + "usr" + File.separator + "servers" + File.separator + serverName + File.separator + "logs";
+        return Paths.get(logDir, hookLog);
+    }
```

Comment:
this doesn't look like it accounts for possibility that server is running with an explicitly set LOG_DIR or non-default server directory location.

Reply:
I don't believe this is settable with wikth the packaged server code. Spoke with Don over slack about it. 

@hlhoots / @bensonlatibm can you confirm? I don't think you can change the log directly when we package the server up in a jar file. At least the code never afforded for that i don't think.